noir-r1cs/src/sparse_matrix.rs

```diff
@@ -43,6 +43,32 @@ impl<F> SparseMatrix<F> {
         assert!(col < self.cols, "column index out of bounds");
         self.entries.insert((row, col), value);
     }
+
+    /// Return a dense representation of the matrix.
+    fn to_dense(&self) -> Vec<Vec<F>>
+    where
+        F: Clone,
+    {
+        let mut result = vec![vec![self.default.clone(); self.cols]; self.rows];
+        for ((i, j), value) in self.entries.iter() {
+            result[*i][*j] = value.clone();
+        }
+        result
+    }
+
+    /// Pretty print a dense representation of the matrix.
+    pub fn pretty_print(&self)
```

Comment:
In Rust `pretty_print` is commonly implemented using the `Debug` or `Display` trait. This will make it work better with the rest of the debug tooling (e.g. you would be able to `dbg!` print a matrix).